File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findDuplicates = function(nums) {
    nums.sort();
    let ans = [];
    
    for(let i = 1; i < nums.length; i++)
        {
            if(nums[i] === nums[i - 1])
                {
                    ans.push(nums[i]);
                }
        }
    return ans;
};

File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.cs
public class Solution {
    public IList<int> FindDuplicates(int[] nums) {
        var dic = new Dictionary<int, int>();
for(var i = 0; i < nums.Length; i++)
{
    if (!dic.ContainsKey(nums[i]))
        dic.Add(nums[i], 1);
    else
        dic[nums[i]]++;
    
}

var ans = dic.Where(num => num.Value > 1).Select(num => num.Key).ToList();
return ans;
    }
}